frontend-react/src/network/api/History/Reports.ts

```diff
@@ -0,0 +1,30 @@
+import { API } from "../NewApi";
+import ActionResource from "../../../resources/ActionResource";
+
+class RSReport {}
+/* For compile-time type checks while #5892 is worked on */
+interface TempRSReport {
```

Comment:
Naming wise, I really dislike the prefix `Temp` 

If it's meant to be deprecated, you could use 
```typescript
/**
 * @deprecated The method should be replaced soon
 */
 ```
 
Not sure if that will pass our github prechecks though.
 
Don't really have a better recommendation, so it's fine the way it is... just the name is confusing when I see it used on other places.

Reply:
Ooo, is this how I get the strikethrough in IDEs when I use functions?! That's a hot tip. I much prefer that. I've got a few other places I can plug that in our repo too haha